packages/web3-eth/src/validation.ts

```diff
@@ -103,3 +119,121 @@ export function isTransactionCall(value: TransactionCall): boolean {
 export function validateTransactionCall(value: TransactionCall) {
 	if (!isTransactionCall(value)) throw new InvalidTransactionCall(value);
 }
+
+export const validateCustomChainInfo = (transaction: Transaction) => {
+	if (transaction.common !== undefined) {
+		if (transaction.common.customChain === undefined) throw new MissingCustomChainError();
+		if (transaction.common.customChain.chainId === undefined)
+			throw new MissingCustomChainIdError();
+		if (
+			transaction.chainId !== undefined &&
+			transaction.chainId !== transaction.common.customChain.chainId
+		)
+			throw new ChainIdMismatchError({
+				txChainId: transaction.chainId,
+				customChainId: transaction.common.customChain.chainId,
+			});
+	}
+
+};
+
+export const validateChainInfo = (transaction: Transaction) => {
+	if (
+		transaction.common !== undefined &&
+		transaction.chain !== undefined &&
+		transaction.hardfork !== undefined
+	) {
+		throw new CommonOrChainAndHardforkError();
+	}
+	if (
+		(transaction.chain !== undefined && transaction.hardfork === undefined) ||
+		(transaction.hardfork !== undefined && transaction.chain === undefined)
+	)
+		throw new MissingChainOrHardforkError({
+			chain: transaction.chain,
+			hardfork: transaction.hardfork,
+		});
+};
+
+export const validateLegacyGas = (transaction: Transaction<HexString>) => {
+	if (
+		// This check is verifying gas and gasPrice aren't less than 0.
+		// transaction's number properties have been converted to HexStrings.
+		// JavaScript doesn't handle negative hex strings e.g. -0x1, but our
+		// numberToHex method does. -0x1 < 0 would result in false, so we must check if
+		// hex string is negative via the inclusion of -
+		transaction.gas === undefined ||
+		transaction.gasPrice === undefined ||
+		transaction.gas.startsWith('-') ||
+		transaction.gasPrice.startsWith('-')
```

Comment:
We can validate this with `uint` type using `web3-validator`

Reply:
Shouldn't [these](https://github.com/ChainSafe/web3.js/blob/ddecdde7a2f11b8e4f969ec985e5c2c6fefed632/packages/web3-validator/test/fixtures/validation.ts#L68) tests fail for being signed (negative) ints?

Reply:
Yes you are right it must fail. Created an issue for it. https://github.com/ChainSafe/web3.js/issues/4777